selfdrive/thermald/thermald.py

```diff
@@ -156,6 +158,23 @@ def handle_fan_uno(max_cpu_temp, bat_temp, fan_speed, ignition):
   return new_speed
 
 
+def set_offroad_alert_if_changed(offroad_alert, show_alert, extra_text=None):
+  if prev_offroad_states.get(offroad_alert, None) == show_alert:
+    return
+  prev_offroad_states[offroad_alert] = show_alert
+  set_offroad_alert(offroad_alert, show_alert, extra_text)
+
+
+def startup_allowed(should_start, condition, reason, offroad_alert=None):
```

Comment:
this seems more complicated than it needs to be. can't you just leave the alerts how they are and build a dict wtih all the things we check? then should start is as simple as `should_start = all(startup_conditions.values())` and if it's blocked log the whole dict.

Reply:
I like the condition dict as well, I do think the offroad_alert wrapper should stay to clean up all the prev value checks though.